apps/computeVSD/computeVSD.cpp

```diff
@@ -195,10 +199,41 @@ class ComputeVSD : public CommandLineApplication
 
         VolumePtr output = source->GetOutput();
         output->SetRegions( fivox::FloatVolume::RegionType( vIndex, vSize ));
-        output->SetSpacing( volumeHandler.computeSpacing( ));
 
-        const fivox::AABBf& bbox = vsdLoader->getBoundingBoxSomas();
-        output->SetOrigin( volumeHandler.computeOrigin( bbox.getCenter( )));
+        fivox::AABBf bboxSomas;
+        const auto& somaPositions = vsdLoader->getSomaPositions();
+        for( const auto& position : somaPositions )
+            bboxSomas.merge( position );
+
+        const auto spacing = volumeHandler.computeSpacing();
+        const auto origin = volumeHandler.computeOrigin( bboxSomas.getCenter());
+
+        if( _vm.count( "soma-pixels" ))
+        {
+            const auto& fileName = _vm["soma-pixels"].as< std::string >();
+            std::ofstream file( fileName );
+            file << "# Soma position and corresponding pixel index for each "
+                    "cell, in the following format:\n"
+                 << "#     gid [ posX posY posZ ]: i j\n"
+                 << "# File version: 1\n"
+                 << "# Fivox version: " << fivox::Version::getString()
+                 << std::endl;
+
+            size_t i = 0;
+            const auto& gids = vsdLoader->getGIDs();
+            for( const auto& gid : gids )
+            {
+                const auto pos = somaPositions[i++];
+                file << gid << " " << pos << ": "
+                     << std::floor(( pos[0]- origin[0] ) / spacing[0] ) << " "
+                     << std::floor(( pos[2]- origin[2] ) / spacing[1] )
```

Comment:
origin should always be lower than pos, and therefore the result a positive value, if I'm not mistaken.